File: Prácticas/java/src/deepspace/BetaPowerEfficientSpaceStation.java
/**
 *
 * @author elena
 */

package deepspace;  

public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation {
    private static final float EXTRAEFFICIENCY= 1.2f;
    private Dice dice;
  
    
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice= new Dice();
    }
    
    @Override
    public float fire(){
        float fire= super.fire();
        return dice.extraEfficiency() ? fire*EXTRAEFFICIENCY : fire;
    }
    
    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }
    
}

File: Prácticas/ruby/lib/BetaPowerEfficientSpaceStation.rb
require_relative 'PowerEfficientSpaceStation'
require_relative 'Dice'
require_relative 'BetaPowerEfficientSpaceStationToUI'

module Deepspace
  class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
    @@EXTRAEFFIENCY= 1.2.freeze

    def initialize(station)
      super
      @dice= Dice.new
    end
    
    def fire
      f= super
      if @dice.extraEfficiency
        f *= @@EXTRAEFFICIENCY  
      end
      
      f
    end
    
    def getUIversion
      BetaPowerEfficientSpaceStationToUI.new(self)
    end
  end
end
